server/src/main/java/org/opensearch/indices/replication/common/ReplicationCollection.java

```diff
@@ -70,6 +70,26 @@ public ReplicationCollection(Logger logger, ThreadPool threadPool) {
         this.threadPool = threadPool;
     }
 
+    /**
+     * Starts a new target event for a given shard, fails the given target if this shard is already replicating.
+     * @param target ReplicationTarget to start
+     * @param activityTimeout timeout for entire replication event
+     * @return The replication id
+     */
+    public long startSafe(T target, TimeValue activityTimeout) {
```

Comment:
we should probably key this collection by shardId instead, but would require some refactoring here.  The only real reuse here between SR and recovery is the monitoring feature of this collection.